app/repositories/onlinetesting/OnlineTestEvaluationRepository.scala

```diff
@@ -42,14 +42,21 @@ trait OnlineTestEvaluationRepository extends CommonBSONDocuments with ReactiveRe
   this: ReactiveRepository[ApplicationReadyForEvaluation, _] =>
 
   val phase: Phase
-  val evaluationApplicationStatuses: List[ApplicationStatus]
+  val evaluationApplicationStatuses: Set[ApplicationStatus]
   val evaluationProgressStatus: ProgressStatus
   val expiredProgressStatus: ProgressStatus
 
   implicit val applicationBSONReader: BSONDocumentReader[ApplicationReadyForEvaluation]
 
   val nextApplicationQuery: String => BSONDocument
 
+  def validEvaluationPhaseStatuses(phase: ApplicationStatus): Set[ApplicationStatus] = {
+    val statusesToIgnore = List(ApplicationStatus.PHASE1_TESTS_FAILED, ApplicationStatus.PHASE2_TESTS_FAILED)
+    ApplicationStatus.values.filter(s =>
+      s >= phase && s < ApplicationStatus.PHASE3_TESTS_PASSED)
+      .filterNot(statusesToIgnore.contains(_))
+  }
```

Comment:
Will include this in the PR for video re-evaluation as it contains the same code block